### PR TITLE
WT-3535 Disable compiler warnings for clang, so ASAN build passes.

### PR DESCRIPTION
**Pull request by agorrod**

### CODE REVIEW

**Comment by agorrod** [comment]
@michaelcahill could you review this for me please? Note that I've opened WT-3536 for re-enabling these warnings.

**Comment by agorrod** [comment]
FWIW: The test failure happening here is the test_timestamp07 timeout.

File: build_posix/aclocal/strict.m4 [hunk 1]
```diff
@@ -90,6 +90,12 @@ AC_DEFUN([AM_CLANG_WARNINGS], [
 	w="$w -Wno-reserved-id-macro"
 	w="$w -Wno-zero-length-array"
 
+	# Added when switching to building with clang 3.4.2
+	w="$w -Wno-parentheses-equality"
+	w="$w -Wno-self-assign"
+	w="$w -Wno-unreachable-code"
+	w="$w -Wno-unreachable-code-return"
```
Comment (michaelcahill) [inline build_posix/aclocal/strict.m4]: This change is sufficient for margay and kodkod.  To get the build working with the version of clang on tinderbox, we'd also need:
```
        w="$w -Wno-array-bounds"
```

**Comment by agorrod** [comment]
Thanks @michaelcahill I've added the -Warray-bounds exclusion, I'm going to go ahead and merge

**Pull request merged**